<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>792</width>
    <height>318</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>BitSet</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Value (64 bit):</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLineEdit" name="_valueLineEdit">
        <property name="inputMask">
         <string>BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB</string>
        </property>
        <property name="text">
         <string>0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="Line" name="line">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <property name="tabBarAutoHide">
       <bool>false</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Statistics</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout_5">
        <item row="0" column="0">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>all():</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="_allLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_5">
          <property name="text">
           <string>none():</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="_noneLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_6">
          <property name="text">
           <string>any():</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="_anyLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_15">
          <property name="text">
           <string>count():</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QLabel" name="_countLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="label_17">
          <property name="text">
           <string>size():</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="_sizeLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3">
       <attribute name="title">
        <string>Convesions</string>
       </attribute>
       <layout class="QFormLayout" name="formLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>toString():</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="_toStringLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>toUlong():</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLabel" name="_toUlongLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>toUllong():</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLabel" name="_toUllongLabel">
          <property name="font">
           <font>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>-</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_4">
       <attribute name="title">
        <string>Access</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0">
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QSpinBox" name="_setBitSpinBox">
              <property name="maximum">
               <number>63</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="_setBitPushButton">
              <property name="text">
               <string>set</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QSpinBox" name="_resetBitSpinBox">
              <property name="maximum">
               <number>63</number>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QPushButton" name="_resetBitPushButton">
              <property name="text">
               <string>reset</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QSpinBox" name="_flipBitSpinBox">
              <property name="maximum">
               <number>63</number>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QPushButton" name="_flipBitPushButton">
              <property name="text">
               <string>flip</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QSpinBox" name="_testBitSpinBox">
              <property name="maximum">
               <number>63</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="_tetstBitPushButton">
              <property name="text">
               <string>test</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>=</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignCenter</set>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="_testBitValueLabel">
              <property name="font">
               <font>
                <bold>true</bold>
               </font>
              </property>
              <property name="text">
               <string>-</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QSpinBox" name="_modifyBitSpinBox">
              <property name="maximum">
               <number>63</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>=</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="_modifyBitValueComboBox">
              <item>
               <property name="text">
                <string>True</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>False</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="_modifyBitPushButton">
              <property name="text">
               <string>[]</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>Bitwise Operators</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_3">
        <item>
         <layout class="QFormLayout" name="formLayout_4">
          <item row="0" column="0">
           <widget class="QPushButton" name="_andPushButton">
            <property name="text">
             <string>&amp;&amp;</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="_andValueLineEdit">
            <property name="inputMask">
             <string>BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB</string>
            </property>
            <property name="text">
             <string>0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="_orPushButton">
            <property name="text">
             <string>|</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QLineEdit" name="_orValueLineEdit">
            <property name="inputMask">
             <string>BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB</string>
            </property>
            <property name="text">
             <string>0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="_xorPushButton">
            <property name="text">
             <string>^</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLineEdit" name="_xorValueLineEdit">
            <property name="inputMask">
             <string>BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB BBBB</string>
            </property>
            <property name="text">
             <string>0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000 0000</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="_notPushButton">
            <property name="text">
             <string>~</string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_5">
       <attribute name="title">
        <string>Shift</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QFormLayout" name="formLayout_2">
          <item row="0" column="0">
           <widget class="QSpinBox" name="_shiftLeftPosSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QPushButton" name="_shiftLeftPushButton">
            <property name="text">
             <string>left</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QSpinBox" name="_shiftRightPosSpinBox">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>64</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="_shiftRightPushButton">
            <property name="text">
             <string>right</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <tabstops>
  <tabstop>_valueLineEdit</tabstop>
  <tabstop>tabWidget</tabstop>
  <tabstop>_setBitSpinBox</tabstop>
  <tabstop>_setBitPushButton</tabstop>
  <tabstop>_resetBitSpinBox</tabstop>
  <tabstop>_resetBitPushButton</tabstop>
  <tabstop>_flipBitSpinBox</tabstop>
  <tabstop>_flipBitPushButton</tabstop>
  <tabstop>_testBitSpinBox</tabstop>
  <tabstop>_tetstBitPushButton</tabstop>
  <tabstop>_modifyBitSpinBox</tabstop>
  <tabstop>_modifyBitValueComboBox</tabstop>
  <tabstop>_modifyBitPushButton</tabstop>
  <tabstop>_andPushButton</tabstop>
  <tabstop>_andValueLineEdit</tabstop>
  <tabstop>_orPushButton</tabstop>
  <tabstop>_orValueLineEdit</tabstop>
  <tabstop>_xorPushButton</tabstop>
  <tabstop>_xorValueLineEdit</tabstop>
  <tabstop>_notPushButton</tabstop>
  <tabstop>_shiftLeftPosSpinBox</tabstop>
  <tabstop>_shiftLeftPushButton</tabstop>
  <tabstop>_shiftRightPosSpinBox</tabstop>
  <tabstop>_shiftRightPushButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>_valueLineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>onValueChanged()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>443</x>
     <y>21</y>
    </hint>
    <hint type="destinationlabel">
     <x>465</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_setBitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onSetPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>108</x>
     <y>97</y>
    </hint>
    <hint type="destinationlabel">
     <x>90</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_resetBitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onResetPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>149</x>
     <y>123</y>
    </hint>
    <hint type="destinationlabel">
     <x>158</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_flipBitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onFlipPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>169</x>
     <y>147</y>
    </hint>
    <hint type="destinationlabel">
     <x>130</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_tetstBitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onTestPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>124</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>142</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_modifyBitPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onModifyPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>188</x>
     <y>222</y>
    </hint>
    <hint type="destinationlabel">
     <x>373</x>
     <y>37</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_andPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onAndPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>9</x>
     <y>5</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_orPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onOrPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>131</y>
    </hint>
    <hint type="destinationlabel">
     <x>34</x>
     <y>-3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_xorPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onXorPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>98</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_notPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onNotPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>188</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>50</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_shiftLeftPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onLeftShitPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>105</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>_shiftRightPushButton</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>onRightShitPushButtonClicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>128</x>
     <y>127</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>46</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onValueChanged()</slot>
  <slot>onSetPushButtonClicked()</slot>
  <slot>onResetPushButtonClicked()</slot>
  <slot>onFlipPushButtonClicked()</slot>
  <slot>onTestPushButtonClicked()</slot>
  <slot>onModifyPushButtonClicked()</slot>
  <slot>onAndPushButtonClicked()</slot>
  <slot>onOrPushButtonClicked()</slot>
  <slot>onXorPushButtonClicked()</slot>
  <slot>onNotPushButtonClicked()</slot>
  <slot>onLeftShitPushButtonClicked()</slot>
  <slot>onRightShitPushButtonClicked()</slot>
 </slots>
</ui>
